test/circuit_breaker_test.rb

```diff
@@ -20,10 +21,15 @@ def test_acquire_yield_when_the_circuit_is_closed
   end
 
   def test_acquire_raises_circuit_open_error_when_the_circuit_is_open
-    open_circuit!
+    Semian.logger = Logger.new($stderr)
```

Comment:
It should be possible to use a StringIO as the logger and inspect it directly without any fd tricks.